temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -2476,4 +2481,22 @@ private StateMachine<ChildWorkflowData> getChildWorkflow(long initiatedEventId)
     }
     return child;
   }
+
+  /**
+   * @return true if the activity state is allowing heartbeats and other updates from the activity
+   *     worker
+   */
+  private boolean isActivityInFlightState(StateMachines.State activityState) {
```

Comment:
Could make this `throwIfActivityNotInFlight` and dedupe the exception throwing by moving it in here